plugins/outputs/bigquery/bigquery.go

```diff
@@ -187,7 +188,17 @@ func (s *BigQuery) newCompactValuesSaver(m telegraf.Metric) (*bigquery.ValuesSav
 		return nil, fmt.Errorf("serializing tags: %w", err)
 	}
 
-	fields, err := json.Marshal(m.Fields())
+	rawFields := make(map[string]interface{}, len(m.FieldList()))
+	for _, field := range m.FieldList() {
+		if fv, ok := field.Value.(float64); ok {
+			// JSON does not support these special values
+			if math.IsNaN(fv) || math.IsInf(fv, 0) {
+				continue
```

Comment:
Maybe we can put a debug message here? Just in case people wonder why their fields are missing...